src/user.ts

```diff
@@ -0,0 +1,285 @@
+import {
+  CommunityContractPeople,
+  fetchBalancesForAddress,
+  fetchContract,
+  fetchUsers,
+  UserBalance,
+} from "verto-cache-interface";
+import { OrderInterface, TransactionInterface, UserInterface } from "./faces";
+import Arweave from "arweave";
+import Utils from "./utils";
+import Token from "./token";
+
+export default class User {
+  private arweave: Arweave;
+  private cache: boolean;
+  private utils: Utils;
+  private token: Token;
+
+  /**
+   *
+   * @param arweave Arweave instance.
+   * @param cache Use the Verto cache.
+   * @param utils Utils submodule.
+   * @param token Token submodule.
+   */
+  constructor(arweave: Arweave, cache: boolean, utils: Utils, token: Token) {
+    this.arweave = arweave;
+    this.cache = cache;
+    this.utils = utils;
+    this.token = token;
+  }
+
+  /**
+   * Fetches the user info for a given wallet address or username.
+   * @param input User wallet address or username.
+   * @returns The user's data such as name & image, or undefined.
+   */
+  async getUser(input: string): Promise<UserInterface | undefined> {
+    let allUsers: CommunityContractPeople[];
+
+    if (this.cache) allUsers = await fetchUsers();
+    else {
+      const contract = await this.utils.getState(this.utils.COMMUNITY_CONTRACT);
+      allUsers = contract.people;
+    }
+
+    return allUsers.find(
+      (user) => user.username === input || user.addresses.includes(input)
+    );
+  }
+
+  /**
+   * Fetches the assets (listed on Verto) for a given wallet address.
+   * @param address User wallet address.
+   * @returns List of asset ids, balances, names, tickers, & logos.
+   */
+  async getBalances(address: string): Promise<UserBalance[]> {
+    if (!this.cache) {
+      const balances: UserBalance[] = [];
+      const listedTokens = await this.token.getTokens();
+
+      for (const token of listedTokens) {
+        const tokenState = await this.utils.getState(token.id);
+
+        if (tokenState?.balances?.[address]) {
+          balances.push({
+            contractId: token.id,
+            name: token.name,
+            ticker: token.ticker,
+            logo: this.utils.getPSTSettingValue("communityLogo", tokenState),
+            balance: tokenState.balances[address],
+            userAddress: address,
+          });
+        }
+      }
+
+      return balances;
+    } else return await fetchBalancesForAddress(address);
+  }
+
+  /**
+   * Fetches the orders for a given wallet address
+   * @param address User wallet address
+   * @returns List of orders
+   */
+  async getOrders(address: string): Promise<OrderInterface[]> {
+    // get clob contract state
+    const clobContractState: {
```

Comment:
I believe this is the same interface I mentioned above. Isolated it in its own interface is better, re-usability and readability.

Reply:
Ah got it now